nixos/modules/system/boot/loader/grub/grub.nix

```diff
@@ -236,6 +237,30 @@ in
         '';
       };
 
+      extraGrubInstallArgs = mkOption {
+        default = [ ];
+        example = [ "--modules=nativedisk ahci pata part_gpt part_msdos diskfilter mdraid1x lvm ext2" ];
```

Comment:
From this, it's not immediately clear to me whether separate arguments go in separate string entries, or if it's allowable to have several options together e.g.:

```[ "--boot-directory=/mnt/usb/bugbios --force --allow-floppy /dev/loop0" ]```

or must it be:

```[ "--boot-directory=/mnt/usb/bugbios" "--force" "--allow-floppy /dev/loop0" ]```

?

Reply:
Additionally, it's not clear if ordering is respected, this may be assumed by a user, especially in the first case, where he or she might want to do something like:

```{enableAhci ? true}:
  {
    achiStr = "" ++ lib.optionalString "ahci";
    extraGrubInstallArgs = [ "--modules=nativedisk" ahciStr "pata part_gpt part_msdos diskfilter mdraid1x lvm ext2" ]
  }
```

(Reading down, it looks like the arguments are just passed to the perl system command, so ordering is respected, and arguments can be combined and split.  So we are dealing with the first case.)

Reply:
> whether separate arguments go in separate string entries, or if it's allowable to have several options together

Separate `grub-install` arugments go into separate string entries. [Example for passing modules this way.](
https://lists.gnu.org/archive/html/help-grub/2012-03/msg00085.html).

The perl script uses [`system PROGRAM LIST`](https://perldoc.perl.org/functions/system.html) to spawn `grub-install`, which explains

> If there is more than one argument in LIST, or if LIST is an array with more than one value, starts the program given by the first element of the list with arguments given by the rest of the list. If there is only one scalar argument, the argument is checked for shell metacharacters, ...

We pass a `LIST` to it, so the individual elements go straight to `execve()`. 

I will add this to `description`:

```
The list elements are passed directly as <literal>argv</literal>
arguments to the <literal>grub-install</literal> program, in order.
````

---

Just for extra clarity for anyone reading, this means that 

> `extraGrubInstallArgs = [ "--modules=nativedisk" ahciStr "pata part_gpt part_msdos diskfilter mdraid1x lvm ext2" ]`

from your example is illegal, because grub expects `--modules=space separated list of modules` to be a single argv argument.